pkg/sitewise/api/list_associated_assets.go

```diff
@@ -9,45 +9,91 @@ import (
 	"github.com/grafana/iot-sitewise-datasource/pkg/framer"
 	"github.com/grafana/iot-sitewise-datasource/pkg/models"
 	"github.com/grafana/iot-sitewise-datasource/pkg/sitewise/client"
-	"github.com/grafana/iot-sitewise-datasource/pkg/util"
 )
 
 func ListAssociatedAssets(ctx context.Context, client client.SitewiseClient, query models.ListAssociatedAssetsQuery) (*framer.AssociatedAssets, error) {
 
 	var (
 		hierarchyId        *string
 		traversalDirection *string
-		assetId            *string = util.GetAssetId(query.BaseQuery)
 		results            []*iotsitewise.AssociatedAssetsSummary
 	)
 
-	// Recursively load children
-	if query.LoadAllChildren {
-		asset, err := client.DescribeAsset(&iotsitewise.DescribeAssetInput{
-			AssetId: assetId,
-		})
+	seenAssetIds := make(map[string]bool)
```

Comment:
Neat Go fact: you could also use `map[string]struct{}` to make a set. I think it's technically slightly better for memory (though on this scale its more of a fun fact that real savings)